### PR TITLE
Amir/ginkgo.master.feature/username replacement api

### PR DESCRIPTION
**Pull request by amirtds**
New API replaces all copies of the username in LMS the desired username. Requires users to be in the username_replacement_admin group. It should only be run as a larger job to update usernames across all services, otherwise, the system will be left in a broken state for those users.
We received a request from ASU to change a username, this can be a solution for the famous change my username request. I tested it and it worked. Only issue I saw was old discussion thread made by my user still has my old username.
https://appsembler.atlassian.net/browse/BLACK-680?atlOrigin=eyJpIjoiY2YzZTViNDA2ODNlNGM3N2I2OTQ5ZmQ5MTkxZmExYTciLCJwIjoiaiJ9

### CODE REVIEW

File: openedx/core/djangoapps/user_api/accounts/views.py [hunk 1]
```diff
@@ -243,3 +255,161 @@ def post(self, request, username):
...
+            "failed_replacements": [
+                {"old_username_2": "new_username_2"}
+            ]
+        }
+
+        TODO: Determine if we need an audit trail outside of logging and API response.
+        """
+
+        # (model_name, column_name)
+        MODELS_WITH_USERNAME = (
```
Comment (bryanlandia) [inline openedx/core/djangoapps/user_api/accounts/views.py]: @amirtds We need to confirm that any add-on packages used by the customer don't introduce models with a username field.  For example, UW RG student statistics does use the username directly.  We should also make sure this doesn't need to do anything for anonymized user ids for edx-submissions (used for ORA, SGA etc.).

Comment (bryanlandia) [reply openedx/core/djangoapps/user_api/accounts/views.py]: Seems like any models using `username` that should match the `User` object should have a plugin entry point to declare that, and then make this tuple configurable by that.

**Comment by bryanlandia** [comment]
@amirtds Also have a look at the CI failures, too, before merging.

**Comment by amirtds** [comment]
Thanks, @bryanlandia yes we need to make sure any new add-on that uses username make it to `MODELS_WITH_USERNAME ` 
I cleaned up Travis errors

**Pull request merged**